numpy/ma/core.py

```diff
@@ -3710,8 +3712,19 @@ def __str__(self):
                 # convert to object array to make filled work
                 names = self.dtype.names
                 if names is None:
-                    res = self._data.astype("O")
-                    res.view(ndarray)[m] = f
+                    data = self._data
+                    mask = m
+                    # For big arrays, to avoid a costly conversion to the
+                    # object dtype, extract the corners before the conversion.
+                    for axis in range(self.ndim):
+                        if data.shape[axis] > self._print_width:
+                            ind = np.int(self._print_width / 2)
```

Comment:
Note that we always use the Python 3 meaning of `/` in order to be able to use the same source code for both Python 2 and 3.